client.go

```diff
@@ -55,12 +52,28 @@ func NewClient(deployFile string, targetName TargetName, isLocalTest bool) (*Mul
 		return nil, err
 	}
 
+	appDir := path.Dir(deployFile)
+
 	// Create all SingleServerClients
 	clients := []Client{}
 	for _, target := range app.Targets(targetName) {
 		localPort := target.Base
+		var serverChannel ServerChannel
+
 		if !isLocalTest {
-			localPort = setupChannel(target.Base, target.SshPort, target.Ssh)
+			localPort = getFreeLocalPort()
+			serverChannel, err = NewSshChannel(
+				target.Base,
+				localPort,
+				target.SshPort,
+				target.Ssh,
+				newCommandRunner(appDir),
+			)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			serverChannel = NewLocalChannel(newCommandRunner(appDir))
```

Comment:
If we can specify arbitrary base ports, do we need local channel for integration tests?  We should be able to run everything without port conflicts?  I'm just worried about increasing codepath differences between tested camus and real camus

(I like separating out interfaces though, regardless, as you've done)